### PR TITLE
Provide login method and authorize all the other actions

### PR DESCRIPTION
**Pull request by dsounded**
In this request:

1) add login method(`POST /session`)
2)Cover all the other methods into Secured section
3)Add and patch specs for both

### CODE REVIEW

File: app/controllers/SessionController.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    new ApiImplicitParam(name = "password", value = "User's password", required = true, dataType = "String", paramType = "body")
+  ))
+  def create = Action.async(BodyParsers.parse.json) { implicit request =>
+    val responder = new Responder[User]
+
+    responder.authorize(Receptionist.checkList(request)) map { response =>
+      val (body, root, status) = response
+
+      body match {
+        case Left(user)   => Status(status)(Json.obj(root -> Json.toJson(User.jsonUserWriter.writesForAuth(user.asInstanceOf[User]))))
```
Comment (hydro0) [inline app/controllers/SessionController.scala]: case Left(user: User) => Status(status)(Json.obj(root -> Json.toJson(User.jsonUserWriter.writesForAuth(user))))

Comment (dsounded) [inline app/controllers/SessionController.scala]: to long ?)

Comment (hydro0) [inline app/controllers/SessionController.scala]: I mean avoid casting, use pattern matching feature )

Comment (dsounded) [inline app/controllers/SessionController.scala]: Oh, though that something had been wrong here, thanks

**Comment by hydro0** [comment]
Ship it!

**Pull request merged**